net.solarnetwork.node.weather.ibm.wc/src/net/solarnetwork/node/weather/ibm/wc/BasicWCClient.java

```diff
@@ -0,0 +1,264 @@
+	}
+
+	private String getURL(String requestURL, String location, String apiKey, String datumPeriod) {
+
+		return this.baseUrl + "/"
+				+ String.format(requestURL, datumPeriod, location, language, format, units, apiKey);
+	}
+
+	/**
+	 * For the daily forecast, each part of the data (ie. max temp, min temp,
+	 * narrative) is an array, with each entry corresponding with a day of the
+	 * week. The more detailed features in "daypart" such as cloud cover are
+	 * split into day and night, meaning two entries for each day.
+	 */
+	private BigDecimal parseBigDecimal(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return n.decimalValue();
+	}
+
+	private LocalTime parseTimeValue(JsonNode n) {
+		if ( n == null || n.isNull() ) {
+			return null;
+		}
+		//log.info(this.dateFormat.parseDateTime(n.asText()).toLocalTime().toString());
+		return this.dateFormat.parseDateTime(n.asText()).toLocalTime();
+
+	}
+
+	private Date parseDateValue(JsonNode n) {
+		if ( n == null || n.isNull() ) {
+			return null;
+		}
+
+		return this.dateFormat.parseDateTime(n.asText()).toDate();
+
+	}
+
+	private Integer parseInt(JsonNode n) {
+		if ( n.isNull() ) {
+			return null;
+		}
+		return n.asInt();
+	}
+
+	//TODO valid time UTC
+	@Override
+	public Collection<GeneralDayDatum> readDailyForecast(String locationIdentifier, String apiKey,
+			DailyDatumPeriod datumPeriod) {
+
+		if ( locationIdentifier == null ) {
+			return null;
+		}
+		final List<GeneralDayDatum> result = new ArrayList<GeneralDayDatum>();
+		final String url = getURL(this.dailyForecastUrl, locationIdentifier, apiKey,
+				datumPeriod.getPeriod());
+		JsonNode root;
+		try {
+			log.debug("opening IBM Weather API connection at URL {}", url);
+			root = getObjectMapper().readTree(jsonGET(url));
+		} catch ( IOException e ) {
+			log.warn("Error reading WC URL [{}]: {}", url, e.getMessage());
+			return result;
+		}
+
+		for ( int i = 0; i < root.get("sunriseTimeLocal").size(); i++ ) {
+
+			GeneralDayDatum current = new GeneralDayDatum();
+			JsonNode data = root.get("sunriseTimeLocal");
+			if ( data.isArray() ) {
+				current.setSunrise(parseTimeValue(data.get(i)));
+			}
+			data = root.get("sunsetTimeLocal");
+			if ( data.isArray() ) {
+
+				current.setSunset(parseTimeValue(data.get(i)));
+
+			}
+			data = root.get("moonriseTimeLocal");
+			if ( data.isArray() ) {
+				current.setMoonrise(parseTimeValue(data.get(i)));
+			}
+			data = root.get("moonsetTimeLocal");
+			if ( data.isArray() ) {
+				current.setMoonset(parseTimeValue(data.get(i)));
+			}
+			data = root.get("validTimeLocal");
+			if ( data.isArray() ) {
+				current.setCreated(parseDateValue(data.get(i)));
+			}
+			data = root.get("temperatureMax");
+			if ( data.isArray() ) {
+				current.setTemperatureMaximum(parseBigDecimal(data.get(i)));
+			}
+			data = root.get("temperatureMin");
+			if ( data.isArray() ) {
+				current.setTemperatureMinimum(new BigDecimal(data.get(i).asLong()));
```

Comment:
Should this line work like line 163, e.g.

```java
current.setTemperatureMinimum(parseBigDecimal(data.get(i)));
```

I'm not clear why they parse the temperatures differently.